### DISCUSSION TITLE
Closing windows using a block

### DISCUSSION THREAD

**Discussion by LeafiDev** [TurboWarp/extensions#900] [Q&A] [answered]
Not sure how it works but when I try to close a window that is opened by TW it doesn't work. what do I need to do to fix it?
![image](https://github.com/TurboWarp/extensions/assets/83046401/b1c7ccbb-1702-4282-aaf9-8ccd7655d3f2)

**Comment by GarboMuffin** [comment]
post more of the extension or try googling that error, it's from your browser

**Reply by ghost** [reply]
Why do you type in all lowercase?

**Reply by GarboMuffin** [reply]
this is the internet, that's just how people talk sometimes

**Reply by LeafiDev** [reply]
I'll be going back to it later. I'll post the extension itself when I get back to it.

**Comment by CubesterYT** [comment, marked as answer]
Im not sure what you are doing, but if you use window.close() it tries to close the main TW window, which it doesn't have permission to do so due to security. It can only close other windows that it creates (child windows of the main window). After my extension, [Window Controls](https://github.com/TurboWarp/extensions/pull/708) gets added, I will be working on an update to make sub windows, of which will show the code to closing sub windows, of which permissions work. Note that window.close() doesn't work on normal tabs only, if its an app or a file like html (basically packaged), it closes fine.

**Reply by LeafiDev** [reply]
If you're doing window controls as an extension then I'll be fine with not doing an extension. Thank you for the rundown.

**Reply by CubesterYT** [reply]
No problem.
